test/message.integration.spec.ts

```diff
@@ -14,8 +14,12 @@ chai.use(chaiAsPromised);
 const { expect } = chai;
 
 const isWin = process.platform === 'win32';
-const isOSX = process.platform === 'darwin';
-const isCI = process.env['CI'] === 'true';
+const isLinux = process.platform === 'linux';
+const isDarwinArm64 = process.platform === 'darwin' && process.arch === 'arm64';
+const isLinuxArm64 = process.platform === 'linux' && process.arch === 'arm64';
+const isCirrusCi = process.env['CIRRUS_CI'] === 'true';
+const usesOctetStream =
+  isLinuxArm64 || isWin || isDarwinArm64 || (isCirrusCi && isLinux);
```

Comment:
I remain uncomfortable with doing the test conditionally - but I don't really see what can be done to improve it. Probably needs a fix in the Rust core so that this concern doesn't leak to consumers of the FFI.

Reply:
The alternative is skip the test completely.

Having the test has value, and the fact that is shows discrepancy between os's/arches means we have a fighting chance of fixing it, there is a tracking issue of the rust core, I'll use this as a show case / test bed

Reply:
the other alternative is leave it as is, and contributors get a sub optimal experience on some arches, as they can't run the repo, and end users get a shitty experience because we don't solve the root issue, which is either a rust core either or just an operating system / arch oddity and all users of pact need guidance on it. 

It doesn't feel like, and it smells, but that pong is enough to get it cleaned up. 

I can only shave so many yaks in a day tho buddy

Reply:
> Probably needs a fix in the Rust core so that this concern doesn't leak to consumers of the FFI.

If it is possible to fix in the core, it's like docker for mac and windows not being able to use localhost to talk to each other

https://docs.docker.com/desktop/networking/#use-cases-and-workarounds-for-all-platforms

maybe it's just one of those things we don't wish to be, but just _is_

Reply:
Yeah. This one is more of a comment than something you can do anything about.